core/local/analysis.js

```diff
@@ -90,78 +108,10 @@ function analyseEvents (events /*: LocalEvent[] */, pendingChanges /*: LocalChan
-            localChange.fileMoveIdenticalOffline(e) ||
-            localChange.fileAddition(e)
-          )
-          break
-        case 'addDir':
-          changeFound(
-            localChange.includeAddDirEventInDirMove(sameInodeChange, e) ||
-            localChange.dirMoveFromUnlinkAdd(sameInodeChange, e) ||
-            localChange.dirRenamingCaseOnlyFromAddAdd(sameInodeChange, e) ||
-            localChange.dirMoveIdenticalOffline(e) ||
-            localChange.dirAddition(e)
-          )
-          break
-        case 'change':
-          changeFound(
-            localChange.includeChangeEventIntoFileMove(sameInodeChange, e) ||
-            localChange.fileMoveFromFileDeletionChange(sameInodeChange, e) ||
-            localChange.fileMoveIdentical(sameInodeChange, e) ||
-            localChange.fileUpdate(e)
-          )
-          break
-        case 'unlink':
-          {
-            const moveChange /*: ?LocalFileMove */ = localChange.maybeMoveFile(sameInodeChange)
-            /* istanbul ignore next */
-            if (moveChange) {
-              // TODO: Pending move
-              panic({path: e.path, moveChange, event: e},
-                'We should not have both move and unlink changes since ' +
-                'checksumless adds and inode-less unlink events are dropped')
-            }
-          }
-          changeFound(
-            localChange.fileMoveFromAddUnlink(sameInodeChange, e) ||
-            localChange.fileDeletion(e) ||
-            withChangeByPath(e, samePathChange => (
-              localChange.convertFileMoveToDeletion(samePathChange) ||
-              localChange.ignoreFileAdditionThenDeletion(samePathChange)
-              // Otherwise, skip unlink event by multiple moves
-            ))
-          )
-          break
-        case 'unlinkDir':
-          {
-            const moveChange /*: ?LocalDirMove */ = localChange.maybeMoveFolder(sameInodeChange)
-            /* istanbul ignore next */
-            if (moveChange) {
-              // TODO: pending move
-              panic({path: e.path, moveChange, event: e},
-                'We should not have both move and unlinkDir changes since ' +
-                'non-existing addDir and inode-less unlinkDir events are dropped')
-            }
-          }
-          changeFound(
-            localChange.dirMoveFromAddUnlink(sameInodeChange, e) ||
-            localChange.dirDeletion(e) ||
-            withChangeByPath(e, samePathChange => (
-              localChange.ignoreDirAdditionThenDeletion(samePathChange) ||
-              localChange.convertDirMoveToDeletion(samePathChange)
-            ))
-          )
-          break
-        default:
-          throw new TypeError(`Unknown event type: ${e.type}`)
-      }
+      const result = analyseEvent(e, changesFound)
+      if (result == null) continue // No change was found. Skip event.
+      if (result === true) continue // A previous change was transformed. Nothing more to do.
```

Comment:
```js
if(APreviousChangeWasTransformed(result)) continue;

// […]

function APreviousChangeWasTransformed(arr) {
  return arr === true;
}
```

Reply:
I expect this part to change again in a future PR, so better avoid premature refactoring.